src/TraceEvent/EventPipe/EventPipeEventSource.cs

```diff
@@ -238,83 +317,41 @@ unsafe class EventPipeEventMetaData
         /// </summary>
         public EventPipeEventMetaData(PinnedStreamReader reader, int length, int fileFormatVersionNumber, int pointerSize, int processId)
         {
-            StreamLabel eventDataEnd = reader.Current.Add(length);
-
+            // Get the event record and fill in fields that we can without deserializing anything.  
             _eventRecord = (TraceEventNativeMethods.EVENT_RECORD*)Marshal.AllocHGlobal(sizeof(TraceEventNativeMethods.EVENT_RECORD));
             ClearMemory(_eventRecord, sizeof(TraceEventNativeMethods.EVENT_RECORD));
 
-            if (pointerSize == 4)
-                _eventRecord->EventHeader.Flags = TraceEventNativeMethods.EVENT_HEADER_FLAG_32_BIT_HEADER;
-            else
-                _eventRecord->EventHeader.Flags = TraceEventNativeMethods.EVENT_HEADER_FLAG_64_BIT_HEADER;
+            _eventRecord->EventHeader.Flags = (ushort)(pointerSize * 8);
```

Comment:
Please don't set the flags this way (I prefer the original).    The issue is that the code is now COMPLETELY obscure (you have to know the specific values of EVENT_HEADER_FLAG_*_BIT_HEADER, to understand why this work.    This is the kind of thing you let optimizer do.  

But more to the point, it is my JOB to do performance investigations/improvements and the very first thing I tell people, is that generally speaking 90+% of all your code is NOT performance sensitive and you should be writing your code to maximize clarity, and maintainability and other metrics.   Only in that last < 10% (it is usually more like 1%) of your code does it matter.   

So is this the hot 10%?    The answer is easy: no.   This code only gets called when metadata events are seen, and those should be probably less than 1% of the trace.   Thus without even doing a measurement I can tell you that this code is NOT the hot 10% and thus we should be optimizing for clarity.   

Now you can debate me whether the 'if' or the ?: is clearer, but I would generally argue that if you have the choice between using a rare construct/pattern and using a common one/cliche, you should use the common one.   'if' is MUCH more common that ?:, and thus should be preferred unless there is a clear advantage (e.g. shrinking 4 lines to 1 is reasonable advantage, but that only works if resulting line is short.   If you have to break it up again, you lose the advantage.

Reply:
I like micro-optimizations, but not that much ;)

For me, it was the just simplest way to write it.

I reverted the change.